### PR TITLE
Tool changes

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Tool vendor is now a shared vendor, has unlimited supplies, and his linked to loadout vendor.
The tool pouche now contains a welder and not a multitool

## Why It's Good For The Game

Tools being in limited supply is dumb, it's absolutly not a balance concern. Welding tool are far more usefull than multi-tool. 

Contraband is a silly mechanic, engie points makes way more sense

## Changelog
:cl:
balance: Tool vendor is now a shared vendor with unlimited supplies
balance: Tool vendor no longer has contraband, the huge tank welder is moved to engie points
expansion: Tool vendor is now linked with loadout system
qol: Tool pouches contains a welding tool rather than a multi-tool
/:cl:

### CODE REVIEW

File: code/__DEFINES/loadout.dm [hunk 1]
```diff
@@ -200,7 +200,7 @@ GLOBAL_LIST_INIT(loadout_linked_vendor, list(
 		/obj/machinery/vending/marineFood,
 		/obj/machinery/vending/MarineMed,
 		/obj/machinery/vending/cigarette,
-		/obj/machinery/vending/cigarette/colony,
```
Comment (BraveMole) [inline code/__DEFINES/loadout.dm:203]: didn't make any sense, how would the loadout vendor connected to a colony vendor

**Comment by TooFewSecrets** [comment]
People will complain, but the autolathe can already print infinite tools from two metal sheets so this is basically just QOL.

**Comment by Walarks** [comment]
Lazy robot mains rejoice!

**Comment by EvilMelonTheSecond** [comment]
What about the High Capacity Industrial Blowtorch we get from hacking?

File: code/game/objects/machinery/vending/vending_types.dm [hunk 2]
```diff
@@ -327,9 +327,17 @@
 	desc = "Tools for tools."
 	icon_state = "tool"
 	icon_deny = "tool-deny"
-	products = list(/obj/item/stack/cable_coil = 10,/obj/item/tool/crowbar = 5,/obj/item/tool/weldingtool = 3,/obj/item/tool/wirecutters = 5,
-					/obj/item/tool/wrench = 5,/obj/item/analyzer = 5,/obj/item/t_scanner = 5,/obj/item/tool/screwdriver = 5)
-	contraband = list(/obj/item/tool/weldingtool/hugetank = 2,/obj/item/clothing/gloves/fyellow = 2, /obj/item/clothing/gloves/yellow = 1)
```
Comment (chesse20) [inline code/game/objects/machinery/vending/vending_types.dm:332]: why remove the contraband weldertool, i see the tool vendor hacked nearly every round and and the huge welders missing

File: code/__DEFINES/loadout.dm (hunk 1 above)
Comment (hyper2snyper) [reply code/__DEFINES/loadout.dm:203]: cuz i stuck em all together in lazyness. good change

File: code/game/objects/machinery/vending/vending_types.dm (hunk 2 above)
Comment (BraveMole) [reply code/game/objects/machinery/vending/vending_types.dm:332]: Put in engie points

**Comment by EvilMelonTheSecond** [comment]
I really like the QoL this brings but the highcap welder being 15 points is too rough. This PR should be atomized

**Comment by Losenis** [comment]
>Tool vendor no longer has contraband, the huge tank welder is moved to engie points

wow the synth nerfs

**Comment by novaepee** [comment]
> 

Easy to put it in synth's locker.

**Comment by novaepee** [comment]
> > Tool vendor no longer has contraband, the huge tank welder is moved to engie points
> 
> wow the synth nerfs

Easy to add in due to this PR. Search up new_marine_vendors.dm, then find /obj/effect/essentials_set/synth, and add in `/obj/item/tool/weldingtool/hugetank`

**Pull request merged**